### PR TITLE
Goom tentacle fix to code broken in earlier commit

### PR DESCRIPTION
**Pull request by glk1001**
The 'tentacle3d.c' code now works properly. (I accidentally broke it in an earlier commit.) Tentacles now look very good with good use of extra color. Turns out 'colorlow' is very important.

Also, the previous issue you mentioned with CVisualizationGoom::AudioData and bad audio sample lines was caused by an incorrect 'audioData' cast in an earlier commit of 'goom_update'. You actually fixed this when you introduced the cleaner thread and buffer handling. And I put back 'getNormalizedData' because I think it adds a nice way to control the width of the audio sample lines.

With the tentacles fixed, I now think Goom looks pretty good.

Tested on an Intel NUC6CAYH and Ubuntu 19.10. Speed is good.

### CODE REVIEW

File: src/Main.h [hunk 1]
```diff
@@ -93,7 +93,7 @@ class ATTRIBUTE_HIDDEN CVisualizationGoom
   GLfloat* m_quadData = nullptr;
 
 #ifdef HAS_GL
-  const static bool g_usePixelBufferObjects = false; // Currently unused, during view change it shortly looks bad with it
+  const static bool g_usePixelBufferObjects = false; // 'true' is supposed to give better performance but it's not obvious.
```
Comment (AlwinEsch) [inline src/Main.h]: A background is that there is a bad picture shortly at the start. But the bigger problem was that when jumping quickly between the windows in Kodi there was screen freeze (for visualization render) in tests.

On both was the screen for visualization not updated:
Bildschirmfoto vom 2020-02-20 20-53-15
Bildschirmfoto von 2020-02-20 20-54-17

We can add them to set over the addon settings by use of OpenGL, by add this to a boolean setting: https://github.com/xbmc/visualization.spectrum/blob/Leia/visualization.spectrum/resources/settings.xml#L30
and on addon a `m_usePixelBufferObjects = kodi::GetSettingsBoolean("use_pixel_buffer_objects");`

File: lib/goom/src/tentacle3d.c [hunk 2]
```diff
@@ -260,15 +259,27 @@ static void pretty_move (PluginInfo *goomInfo, float cycle, float *dist, float *
...
+
+	*color = (*color1) * (*color2) / 256;
+	color++; color1++; color2++;
+	*color = (*color1) * (*color2) / 256;
+	color++; color1++; color2++;
+	*color = (*color1) * (*color2) / 256;
+	color++; color1++; color2++;
+	*color = (*color1) * (*color2) / 256;
+
+  return col;
```
Comment (AlwinEsch) [inline lib/goom/src/tentacle3d.c]: Can you remove the tabs there.

**EDIT:** Or a tab on `uint32_t col;`

**Review by AlwinEsch — COMMENTED** [review]
In the test it worked so far on my Linux, I will check tomorrow on the other OS.

I apologize that it was gone, only previously no audio signal arrived properly in the goom, hence the `circular_buffer` in it now.

Can you also bring a request to the matrix branch, or should I use it after this request (cherry pick)?

**Comment by AlwinEsch** [comment]
This `Merge branch 'Leia' into Leia_tentacle_fix` was wrongly by a branch setting change added, you can kick out them.

File: lib/goom/src/tentacle3d.c (hunk 2 above)
Comment (glk1001) [reply lib/goom/src/tentacle3d.c]: I've made the tabs consistent in tentacle3d.c.

File: src/Main.h (hunk 1 above)
Comment (glk1001) [reply src/Main.h]: I've made 'm_usePixelBufferObjects' use a kodi setting.

**Comment by glk1001** [comment]
Hi, I force pushed the changes you requested including removing the previous Leia merge. Might be easiest if you cherry-pick the final request to the matrix branch.

File: visualization.goom/resources/settings.xml [hunk 3]
```diff
@@ -18,6 +18,10 @@
           </constraints>
           <control type="spinner" format="string" />
         </setting>
+        <setting id="use_pixel_buffer_objects" type="boolean" label="30007" help="30008">
```
Comment (AlwinEsch) [inline visualization.goom/resources/settings.xml:21]: You need then also to add on `resources/language/resource.language.en_gb/strings.po`, e.g. :smile::
```po
msgctxt "#30007"
msgid "Pixel buffer objects"
msgstr ""

msgctxt "#30008"
msgid "To increase render performance"
msgstr ""
```
my english not the best, but something like this then needed.

Comment (glk1001) [reply visualization.goom/resources/settings.xml:21]: Sorry, my mistake, forgot to test previous changes. Added strings to 'strings.po'. All should be good now.

**Comment by glk1001** [comment]
Updated strings.po and tested. Hopefully all is well.

**Comment by AlwinEsch** [comment]
Make the following to remove this `This branch is out-of-date with the base branch`
```sh
git remote add upstream https://github.com/xbmc/visualization.goom
git fetch upstream
git rebase upstream/Leia # inside the Leia_tentacle_fix branch
```
If I press this `Update branch`, then comes again the extra commit.

**Comment by glk1001** [comment]
Hopefully all good now.

**Comment by AlwinEsch** [comment]
From work it is good and ready, only in new setting one part:
```xml
          <dependencies>
            <dependency type="visible">
              <condition on="property" name="IsDefined">HAS_GL</condition>
            </dependency>
          </dependencies>
```
Where I give on Link above to show this only on GL (not on GLES).

I add this by release of them.

**Comment by AlwinEsch** [comment]
Thanks a lot for your work!

**Pull request merged**